src/commands/test/run_test_runner.py

```diff
@@ -37,23 +40,34 @@ async def run_test_runner(
 
     with multiprocessing.Manager() as manager:
         queue = manager.Queue()  # type: ignore
-        reporter = ReporterCoordinator(tests_root, test_subjects, queue)
+        reporter_coordinator = ReporterCoordinator(
+            tests_root, test_subjects, queue, logger
+        )
         setups = [
             (
                 subject,
-                reporter.create_reporter(subject),
+                reporter_coordinator.create_reporter(),
                 include_paths,
             )
             for subject in test_subjects
         ]
 
-        with Pool(multiprocessing.cpu_count()) as pool:
+        with Pool(multiprocessing.cpu_count(), worker_init) as pool:
             result = pool.starmap_async(run_worker, setups)
-            reporter.live_reporting()
-            results = result.get()
-            reporter.report_summary(results)
+            try:
+                reporter_coordinator.live_reporting()
+                return result.get()
+            except KeyboardInterrupt:
+                pool.terminate()
+                pool.join()
+                return []
+
+
+def worker_init():
+    def signal_handler(_signal_num, _frame):
+        return None
 
-        return results
+    signal.signal(signal.SIGINT, signal_handler)
```

Comment:
ctrl+c does not have to guarantee gracefull handling. In my opion we should leave it as it was, this solution introduces unnecessary layer of complexity.

Reply:
Ctrl + c will often be used to stop tests. We must handle it somehow. Currently, Protostar generate a lot of noise when tests are interrupted.